src/app/components/pages/page.component.ts

```diff
@@ -55,9 +55,22 @@ export abstract class PageComponent extends BaseComponent implements OnInit, OnD
 
     this.pageOptionsService.setTransparentNavbarWhenTopScrolled(this.transparentNavbarWhenTopScrolled);
 
+    this.sendAnalyticsEvent();
```

Comment:
Is this really the best place to put this code @bex1 any other ideas or is this OK?

Reply:
I think that the code should be placed in the navigation.service.ts instead.

You can use the router.events subscription in the initService method.

```
this.router.events.subscribe(event => {
      if (event instanceof NavigationStart) {
        console.log('Starting navigation to ' + event.url)
      }

      if (event instanceof NavigationEnd) {
        this.currentUrl = this.router.url;
        if (isPlatformBrowser(this.platformId)) {
          window.scrollTo(0, 0);
        }
        console.log('Navigation ended at ' + this.currentUrl);
      }

      if (event instanceof NavigationCancel) {
        console.log('NavigationCancel');
      }

      if (event instanceof NavigationError) {
        console.log('NavigationError', event);
      }

      if (event instanceof RoutesRecognized) {
        console.log('RoutesRecognized');
      }
    });
```

Reply:
OK, thanks 👍